### PR TITLE
quillcpp: init at 3.3.1

### PR DESCRIPTION
**Pull request by imnotpoz**
## Description of changes

https://github.com/odygrd/quill

(not sure if adding the library to a project and compiling counts as testing the basic functionality of binary files but I did that and it worked)

the patch is a PR I submitted, will remove it once it gets merged into upstream and a new version is released

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -1951,6 +1951,8 @@ with pkgs;
 
   quich = callPackage ../tools/misc/quich { } ;
 
+  quillcpp = callPackage ../by-name/qu/quillcpp/package.nix { };
+
```
Comment (FliegendeWurst) [inline pkgs/top-level/all-packages.nix:1955]:
```suggestion
```

File: pkgs/by-name/qu/quillcpp/package.nix [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    rev = "v${version}";
+    hash = "sha256-aPGAinaws60aGinUD926QgM3Mywym7tnNXJRq1xtXA0=";
+  };
+
+  patches = [
+    (fetchpatch {
+      name = "fix-pkg-config-lib-and-include-paths.patch";
+      url = "https://patch-diff.githubusercontent.com/raw/odygrd/quill/pull/352.patch";
+      hash = "sha256-/p+p6OnGybV86r1jMKj+JbaW/JhwMv6Nbj6dzn4p2sQ=";
+    })
```
Comment (FliegendeWurst) [inline pkgs/by-name/qu/quillcpp/package.nix:28]: Usually such small patches would be copied into nixpkgs.

**Review by FliegendeWurst — COMMENTED** [review]
> (not sure if adding the library to a project and compiling counts as testing the basic functionality of binary files but I did that and it worked)

It does! Thank you for packaging quillcpp. I have left some review comments.

By the way, your commits are not associated with your Github account ("No user is associated with the committer email.").

File: maintainers/maintainer-list.nix [hunk 3]
```diff
@@ -7608,6 +7608,13 @@
     github = "jacbart";
     githubId = 7909687;
   };
+  jacekpoz = {
+    name = "Jacek Poziemski";
+    email = "jacekpoz@cock.li";
+    matrix = "@jacekpoz:jacekpoz.pl";
+    github = "jacekpoz";
+    githubId = 64381190;
+  };
```
Comment (FliegendeWurst) [inline maintainers/maintainer-list.nix:7617]: Changes to maintainer-list.nix are usually done in a separate commit.

**Comment by SuperSandro2000** [comment]
dupe of #316404
